src/Ledger/PPUp.lagda

```diff
@@ -92,6 +92,8 @@ data _⊢_⇀⦇_,PPUP⦈_ : PPUpdateEnv → PPUpdateState → Maybe Update →
   PPUpdateCurrent : let open PPUpdateEnv Γ in
     dom (pup ˢ) ⊆ dom (genDelegs ˢ)
     → All (isViableUpdate pparams) (range (pup ˢ))
+    -- I would like to remove the `ˢ` decoration from `<ˢ`, but I believe we need a *strict*
+    -- order type class for that and including such a type class was voted down.
     → (slot + (2 * StabilityWindow)) <ˢ firstSlot (sucᵉ (epoch slot))
```

Comment:
I don't see how or why we would use `StrictToNonStrict` or `NonStrictToStrict` to try to make inference for `<` work with only a (nonstrict) partial order type class available.

Reply:
I tried this approach, but a problem occurs when a type (e.g., `ℕ`) has it's own strict partial order; that order will conflict with the strict partial order derived from the inferred partial order, so we must override the inferred `_<_` with a decorated symbol (e.g., `_<ℕ_`)... and now we're back to needing annotated relation symbols... even worse, we're now annotating relations defined on primitive types, instead of annotating the relations of special "user defined" types.

I really think it makes more sense, would be easier and cleaner, to include a separate `StrictPartialOrder` type class in the HasOrder module.